### PR TITLE
Support for Extensions 2.0

### PR DESCRIPTION
**Pull request by bilalhashmi-dt**
Added functionality for List, ListAllByName, Get, Delete
added tests for list, get

### CODE REVIEW

File: dynatrace/environment_v2/extensions.py [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+    def list(self) -> PaginatedList["MinimalExtension"]:
+        """ Lists all the extensions 2.0 available in your environment
+
+        :return: a list of MinimalExtension objects
+        """
+        return PaginatedList(MinimalExtension, self.__http_client, target_url=self.ENDPOINT, list_item="extensions")
+
+
+    def listByName(self, extension_name: str) -> PaginatedList["MinimalExtension"]:
```
Comment (dlopes7) [inline dynatrace/environment_v2/extensions.py:43]: This is `list_versions`, not `listByName`, python also does not use camelCase for method names, always lower case with underscores

File: dynatrace/environment_v2/extensions.py [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+        """ Lists all the extensions 2.0 by specified name in your environment
+
+        :param extension_name: the name of the extension 2.0
+
+        :return: a list of MinimalExtension objects
+        """
+        return PaginatedList(MinimalExtension, self.__http_client, target_url=f"{self.ENDPOINT}/{extension_name}", list_item="extensions")
+
+
+    def listEvents(self, extension_name: str) -> PaginatedList["ExtensionEventDTO"]:
```
Comment (dlopes7) [inline dynatrace/environment_v2/extensions.py:53]: Call this `list_environment_config_events`

File: dynatrace/environment_v2/extensions.py [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+        """ List of the latest extension environment configuration events
+
+        :param extension_name: the name of the extension 2.0
+
+        :return: a list of ExtensionEventDTO object
+        """
+        return PaginatedList(ExtensionEventDTO, self.__http_client, target_url=f"{self.ENDPOINT}/{extension_name}/environmentConfiguration/events", list_item="extensionEvents")
+
+
+    def listEventsByConfigId(self, extension_name: str, config_id: str) -> PaginatedList["ExtensionEventDTO"]:
```
Comment (dlopes7) [inline dynatrace/environment_v2/extensions.py:63]: Call this `list_monitoring_config_events`, and same thing as before, `list_monitoring_config_events(config_id)` method on the `MinimalExtension` instance

File: dynatrace/environment_v2/extensions.py [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+
+        :param extension_name: the name of the requested extension 2.0
+        :param extension_version: the version of the requested extension 2.0
+
+        :return: HTTP response
+        """
+        return self.__http_client.make_request(path=f"{self.ENDPOINT}/{extension_name}/{extension_version}", method="DELETE")
+    
+
+    def getActiveExtensionVersion(self, extension_name: str) -> "ExtensionEnvironmentConfigurationVersion":
```
Comment (dlopes7) [inline dynatrace/environment_v2/extensions.py:97]: This can be `get_environment_config`

File: dynatrace/environment_v2/extensions.py [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+
+        :param extension_name: the name of the requested extension 2.0
+
+        :return: ExtensionEnvironmentConfigurationVersion object
+        """
+        response = self.__http_client.make_request(f"{self.ENDPOINT}/{extension_name}/environmentConfiguration").json()
+        return ExtensionEnvironmentConfigurationVersion(raw_element=response)
+
+
+    def updateActiveExtensionVersion(self, extension_name: str, _version: str):
```
Comment (dlopes7) [inline dynatrace/environment_v2/extensions.py:108]: `put_environment_config` This needs to be similar to other implementations like `SLOs`, where this method receives a `ExtensionEnvironmentConfigurationVersion` instance, and calls the `put()` method of that instance.

File: dynatrace/environment_v2/extensions.py [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+        :param extension_name: the name of the requested extension 2.0
+        :param _version: the change to be put in place and activated
+
+        :return: HTTP response
+        """
+        params = {"version": _version}
+        return self.__http_client.make_request(path=f"{self.ENDPOINT}/{extension_name}/environmentConfiguration", params=params, method="PUT")
+    
+    
+    def deactivateExtension(self, extension_name: str):
```
Comment (dlopes7) [inline dynatrace/environment_v2/extensions.py:120]: `delete_environment_config`

File: dynatrace/main.py [hunk 7]
```diff
@@ -89,6 +90,7 @@ def __init__(
         self.entities: EntityService = EntityService(self.__http_client)
         self.events: EventService = EventService(self.__http_client)
         self.extensions: ExtensionService = ExtensionService(self.__http_client)
+        self.extensionsv2: ExtensionsServiceV2 = ExtensionsServiceV2(self.__http_client)
```
Comment (dlopes7) [inline dynatrace/main.py:93]: `self.extensions_v2`

File: dynatrace/environment_v2/extensions.py (hunk 1 above)
Comment (dlopes7) [reply dynatrace/environment_v2/extensions.py:43]: It also needs to be possible to do 

```
for e in dt.extensions_v2.list():
    for version in e.list_versions(): # as well as for version in dt.extensions_v2.list_versions(e.extension_name)
       print(version
```

So there needs to be a `list_versions` method in the `MinimalExtension` class

File: dynatrace/environment_v2/extensions.py (hunk 2 above)
Comment (dlopes7) [reply dynatrace/environment_v2/extensions.py:53]: Same thing as before as well, it must be possible to call `list_environment_config_events` directly in a `MinimalExtension` instance

File: dynatrace/environment_v2/extensions.py (hunk 4 above)
Comment (dlopes7) [reply dynatrace/environment_v2/extensions.py:97]: Also needs to be possible to call it directly from `MinimalExtension`

**Review by dlopes7 — CHANGES_REQUESTED** [review]
@bilalhashmi-dt This one needs a lot of changes, and this is an important service because it will be extensively used by github.com/dynatrace-oss/dt-cli

Can you enable `Allow edit from maintaners` in this pull request? This way I can collaborate on this as well, here is how to do it:

https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork

**Comment by bilalhashmi-dt** [comment]
Hi @dlopes7 - `Allow edits by maintainers` is checked you should be all set. Thanks so much - these are really helpful comments, I will begin work on them at the earliest

File: dynatrace/environment_v2/extensions.py (hunk 2 above)
Comment (bilalhashmi-dt) [reply dynatrace/environment_v2/extensions.py]: @dlopes7 please confirm you mean this method must be callable directly in a `ExtensionEventDTO` instance

**Comment by bilalhashmi-dt** [comment]
Opened new PR #36 with included changes

**Pull request closed without merging**